plugins/obs-vst/CMakeLists.txt

```diff
@@ -12,46 +12,48 @@ endif()
 add_library(obs-vst MODULE)
 add_library(OBS::vst ALIAS obs-vst)
 
-find_qt(COMPONENTS Widgets)
+find_package(Qt6 REQUIRED Widgets)
 
 target_sources(
   obs-vst
-  PRIVATE obs-vst.cpp
-          VSTPlugin.cpp
+  PRIVATE # cmake-format: sortable
+          $<$<PLATFORM_ID:Darwin>:mac/EditorWidget-osx.mm>
+          $<$<PLATFORM_ID:Darwin>:mac/VSTPlugin-osx.mm>
+          $<$<PLATFORM_ID:Linux,FreeBSD>:linux/EditorWidget-linux.cpp>
```

Comment:
While this would only take effect after the Linux updates have been merged, it would actually be up to debate how "supported" OpenBSD actually is. Without an explicit list of supported features by the platform I'm currently not able to say which parts of OBS Studio are supported in OpenBSD at all.

Reply:
As far as I know, OBS on OpenBSD should operate much the same as on FreeBSD, with some exceptions.
See https://github.com/obsproject/obs-studio/commit/0d222b6b56fa3a49ed3a6604186870dd9f90e88b.

Reply:
In that case I'll have to do another pass over the entire codebase here (and in the Linux update) to explicitly add OpenBSD or revert to a negative check ("not Windows and also not macOS") to lump in any other platform.

Otherwise we'd have to use `PLATFORM_ID`-based generator expressions for Windows and macOS, but a manual `BOOL`-based expression to check for any kind of Linux or Unix.

Reply:
Right. As mentioned, before we seem to have previously (currently?) put Linux, FreeBSD, and OpenBSD together in CMake as being `OS_POSIX` (`cmake/Modules/ObsHelpers.cmake`), but now the new CMake path (`cmake/common/osconfig.cmake`) creates:
* OS_LINUX
* OS_FREEBSD
* OS_OPENBSD

Reply:
Updated to explicitly include all three of those.

Reply:
As an addendum, it looks like the old path actually set _both_ OS_POSIX _and_ OS_LINUX, OS_FREEBSD, and OS_OPENBSD. Is there a reason we omitted OS_POSIX from the new path?

Reply:
`OS_POSIX` originally also included macOS and served as a general "not Windows" catch-all which was already changed when I made macOS a more distinct platform.

This change just continues that to make platform selection explicit across the board.

It also avoids conditionals like `if(OS_POSIX AND NOT OS_FREEBSD)` which seem less clear than `if(OS_LINUX AND OS_OPENBSD)` to me.